### PR TITLE
Rework linear geometry test

### PR DESCRIPTION
**Pull request by sandro-elsweijer**
**_Describe your changes here:_**
Move linear geometry test to a separate file to keep structure from src directory.
Moreover, adapt the linear geometry test so that it tests all element shapes.

**_All these boxes must be checked by the reviewers before merging the pull request:_**

As a reviewer please read through all the code lines and make sure that the code is fully understood, bug free, well-documented and well-structured.

#### General
- [ ] The reviewer executed the new code features at least once and checked the results manually

- [ ] The code follows the [t8code coding guidelines](https://github.com/holke/t8code/wiki/Coding-Guideline)
- [ ] New source/header files are properly added to the Makefiles
- [ ] The code is well documented
- [ ] All function declarations, structs/classes and their members have a proper doxygen documentation
- [ ] All new algorithms and data structures are sufficiently optimal in terms of memory and runtime (If this should be merged, but there is still potential for optimization, create a new issue)

#### Tests
- [ ] The code is covered in an existing or new test case using Google Test

#### Github action

- [ ] The code compiles without warning in debugging and release mode, with and without MPI (this should be executed automatically in a github action)
- [ ] All tests pass (in various configurations, this should be executed automatically in a github action)

  If the Pull request introduces code that is not covered by the github action (for example coupling with a new library):
  - [ ] Should this use case be added to the github action?
  - [ ] If not, does the specific use case compile and all tests pass (check manually)

#### Scripts and Wiki

- [ ] If a new directory with source-files is added, it must be covered by the `script/find_all_source_files.scp` to check the indentation of these files.
- [ ] If this PR introduces a new feature, it must be covered in an example/tutorial and a Wiki article.

#### Licence

- [ ] The author added a BSD statement to `doc/` (or already has one)

### CODE REVIEW

File: test/t8_geometry/t8_geometry_implementations/t8_gtest_geometry_linear.cxx [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+#include <test/t8_geometry/t8_gtest_geometry_macros.hxx>
+#include <gtest/gtest.h>
+#include <t8_eclass.h>
+#include <t8_cmesh.h>
+#include <t8_cmesh/t8_cmesh_examples.h>
+#include <t8_geometry/t8_geometry.h>
+#include <t8_geometry/t8_geometry_implementations/t8_geometry_linear.hxx>
+#include <t8_element.h>
+
+t8_cmesh_t
```
Comment (jmark) [inline test/t8_geometry/t8_geometry_implementations/t8_gtest_geometry_linear.cxx:37]:
```suggestion
static t8_cmesh_t
```

**Review by jmark — CHANGES_REQUESTED** [review]
LGTM! Just a minor suggestion.

**Comment by jmark** [comment]
@sandro-elsweijer Please merge in the latest `main`.

**Review by jmark — APPROVED** [review]

**Comment by sandro-elsweijer** [comment]
The changes were already introduced with the new geometry handler https://github.com/DLR-AMR/t8code/pull/950

**Pull request closed without merging**